src/Dialect/Json.php

```diff
@@ -165,9 +171,14 @@ public function setAttribute($key, $value)
     public function setJsonAttribute($attribute, $key, $value)
     {
         $obj = json_decode($this->{$attribute});
-        $obj->$key = $value;
-        $this->flagJsonAttribute($key, $attribute);
-        $this->{$attribute} = json_encode($obj);
+        if(!is_object($obj))
+            $obj = new stdClass;
+        // only set this if $value isn't null.
+        if(!is_null($value) ) {
```

Comment:
Your right.  I had a very specific use case at the time... but is not longer relevant and no reason to do it that way.